packages/vscode-extension/src/debug/taskTerminal/baseTaskTerminal.ts

```diff
@@ -0,0 +1,47 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as vscode from "vscode";
+
+const ControlCodes = {
+  CtrlC: "\u0003",
+};
+
+export abstract class BaseTaskTerminal implements vscode.Pseudoterminal {
+  protected writeEmitter = new vscode.EventEmitter<string>();
+  onDidWrite: vscode.Event<string> = this.writeEmitter.event;
+  protected closeEmitter = new vscode.EventEmitter<number>();
+  onDidClose?: vscode.Event<number> = this.closeEmitter.event;
+
+  constructor(private taskDefinition: vscode.TaskDefinition) {}
+
+  open(): void {
+    this.do()
+      .then(() => this.stop())
+      .catch((error) => this.stop(error));
+  }
+
+  close(): void {
+    this.stop();
+  }
+
+  handleInput(data: string): void {
+    this.writeEmitter.fire(data === "\r" ? "\r\n" : data);
+    if (data.includes(ControlCodes.CtrlC)) {
```

Comment:
Will some multi-byte Unicode character or control sequences contain '\x03'? If so will it be treated as Ctrl-C?

Reply:
Here the value of ctrl-c is the virtual terminal sequence.
```
Implement to handle incoming keystrokes in the terminal or when an extension calls [sendText](https://code.visualstudio.com/api/references/vscode-api#Terminal.sendText). data contains the keystrokes/text serialized into their corresponding VT sequence representation.
```